MapboxCoreNavigation/RouteController.swift

```diff
@@ -330,6 +332,55 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func checkForFasterRoute(from location: CLLocation) {
+        
+        // If the user does not have much time left on the route, don't check for faster alternatives
+        guard routeProgress.durationRemaining > 600 else { return }
+
+        // If the user is approaching a maneuver, don't check for a faster alternatives
+        guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining > 70 else { return }
+        
+        guard let currentUpcomingManeuver = routeProgress.currentLegProgress.upComingStep else { return }
+        
+        guard let lastLocationTimestamp = lastLocationDate else {
+            lastLocationDate = location.timestamp
+            return
+        }
+
+        // Only check ever 2 minutes for faster route
+        guard location.timestamp.timeIntervalSince(lastLocationTimestamp) > 120 else { return }
+        
+        let options = routeProgress.route.routeOptions
+        let firstWaypoint = Waypoint(coordinate: location.coordinate)
+        if location.course >= 0 {
+            firstWaypoint.heading = location.course
+            firstWaypoint.headingAccuracy = 90
+        }
+        options.waypoints = [firstWaypoint] + routeProgress.remainingWaypoints
+        
+        let durationRemaining = routeProgress.durationRemaining
+        let currentAlertLevel = routeProgress.currentLegProgress.alertUserLevel
+        
+        routeTask?.cancel()
+        routeTask = directions.calculate(options, completionHandler: { [weak self] (waypoints, routes, error) in
+            guard let strongSelf = self else { return }
+            
+            strongSelf.lastLocationDate = nil
+            
+            guard let route = routes?.first else { return }
+            
+            let percentDifference = ((durationRemaining - route.expectedTravelTime) / durationRemaining) * 100
+            
+            // Only use new route if it's at least 10% faster
+            if percentDifference >= 10 {
```

Comment:
This can be simplified algebraically:

```swift
if route.expectedTravelTime <= 0.9 * durationRemaining
```

This is also safer, on the off-chance that we get into this code when the user is already at the destination. With the current code, there would be a division by zero.